pkgs/stdenv/generic/setup.sh

```diff
@@ -1280,6 +1299,11 @@ genericBuild() {
         if [ "$curPhase" = unpackPhase ]; then
             cd "${sourceRoot:-.}"
         fi
+        phase_end_time=$(date '+%s')
+        duration=$((phase_end_time - phase_start_time))
+        if [ -n "${enablePhaseMetrics:-}" ]; then
+            echo "time.$curPhase $duration" >> "$NIX_BUILD_TOP/metrics"
```

Comment:
I wanted it to be unique since both would show up in the metrics file, it might get confusing which one is the duration between two timestamps and which one is the performance metric `real` coming from the more detailed function.